actions/subscribers.go

```diff
@@ -472,9 +471,27 @@ func ExportSubscribers(c *gin.Context) {
 		return
 	}
 
-	go reportSvc.GenerateExportReport(c, report)
+	go func(c context.Context, report *entities.Report) {
+		err := reportSvc.GenerateExportReport(c, report)
+		if err != nil {
+			// report failed
+			report.Status = entities.StatusFailed
+			logger.From(c).WithFields(logrus.Fields{
+				"report": report,
+			}).WithError(err).Errorf("Export failed")
+		}
+
+		// report finished successfully
+		report.Status = entities.StatusDone
 
-	c.JSON(http.StatusOK, report)
+		// update report status
+		err = storage.UpdateReport(c, report)
```

Comment:
The update of the status can be part of GenerateExportReport, like previously, just return an error instead of logging the errors inside the method. Log them here.